src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -143,17 +143,37 @@ private static bool AudienceIsValid(IEnumerable<string> audiences, TokenValidati
                 if (string.IsNullOrWhiteSpace(tokenAudience))
                     continue;
 
-                foreach (string validAudience in validationParametersAudiences)
+                if (validationParametersAudiences is IList<string> audienceList)
                 {
-                    if (string.IsNullOrWhiteSpace(validAudience))
-                        continue;
+                    for (int i = 0; i < audienceList.Count; i++)
+                    {
+                        string validAudience = audienceList[i];
+                        if (string.IsNullOrWhiteSpace(validAudience))
+                            continue;
 
-                    if (AudiencesMatch(validationParameters, tokenAudience, validAudience))
+                        if (AudiencesMatch(validationParameters, tokenAudience, validAudience))
```

Comment:
This code could be in a local function.

Reply:
cline has addressed the PR feedback. I only shared your comment and the line of code.